gsrs-module-substances-core/src/main/java/ix/ginas/models/v1/Substance.java

```diff
@@ -1496,4 +1498,77 @@ public static String toHexString(byte[] bytes) {
 
         return hexString.toString();
     }
+    
+    
+    // tags begin
+
+    public List<String> grabTagTerms() {
+        // This was named "getTagTerms," but I got strange PojoDiff error.
+        // It was a 500 error complaining about equal index and size values
+        // in the list. It occurred when trying to delete a tag *and* persisting.
+        List tagTerms = new ArrayList<String>();
+        for(Keyword k: this.tags){
+            tagTerms.add(k.getValue());
+        }
+        return tagTerms;
+    }
+
+    public Set<String> extractTagTermsFromNamesAsSet() {
+        List<Name> names = this.getAllNames();
+        if (names == null) { return null; }
+        Set<String> tagTerms = new HashSet<String>();
+        for (Name name: names) {
+            String e = name.extractTagTermFromName();
+            if (e != null) {
+                tagTerms.add(e);
+            }
+        }
+        return tagTerms;
+    }
+
+    public List<String> extractDistinctTagTermsFromNames() {
```

Comment:
Eliminate this method entirely. The set one above does everything this does better.

Reply:
The concern I had there was that I'd have to convert the Set into a List  to the  collections.remove(list1, list 2)  ... so why not save downstream coders that step.